src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.MultiAgent.cs

```diff
@@ -833,6 +839,39 @@ private static ulong CurlSendCallback(IntPtr buffer, ulong size, ulong nitems, I
                 return Interop.Http.CURL_READFUNC_ABORT;
             }
 
+            private static void SetLowSpeedLimit(EasyRequest easy, TimeSpan timeout)
+            {
+                // If there's no send/receive timeout on the handler, the defaults will
+                // remain disabled, and we don't need to worry about setting or resetting.
+                if (!easy._handler._hasSendReceiveTimeout)
+                {
+                    return;
+                }
+
+                // We need to ensure the timeout is set appropriately, which may include
+                // disabling a timeout currently set if one was.
+
+                long limit = 0, time = 0; // 0 == disable
+
+                if (timeout != Timeout.InfiniteTimeSpan)
+                {
+                    double secs = timeout.TotalSeconds;
+                    if (secs >= 1)
+                    {
+                        limit = 1;
+                        time = (long)Math.Ceiling(secs);
+                    }
+                    else
+                    {
+                        limit = (long)Math.Ceiling(1 / secs);
+                        time = 1;
+                    }
+                }
+
+                easy.SetCurlOption(Interop.Http.CURLoption.CURLOPT_LOW_SPEED_LIMIT, limit);
+                easy.SetCurlOption(Interop.Http.CURLoption.CURLOPT_LOW_SPEED_TIME, time);
+            }
```

Comment:
It might be useful to explain that SPEED_LIMIT is bytes/sec and SPEED_TIME is how long the transfer rate needs to be below SPEED_LIMIT to abort.